src/app/indexed-db/key-performance-metric-impacts-idb.service.ts

```diff
@@ -0,0 +1,86 @@
+  }
+
+  getAll(): Observable<Array<IdbKeyPerformanceMetricImpact>> {
+    return this.dbService.getAll('keyPerformanceMetricImpact');
+  }
+
+  getById(id: number): Observable<IdbKeyPerformanceMetricImpact> {
+    return this.dbService.getByKey('keyPerformanceMetricImpact', id);
+  }
+
+  addWithObservable(keyPerformanceMetricImpact: IdbKeyPerformanceMetricImpact): Observable<IdbKeyPerformanceMetricImpact> {
+    return this.dbService.add('keyPerformanceMetricImpact', keyPerformanceMetricImpact);
+  }
+
+  deleteWithObservable(id: number): Observable<any> {
+    return this.dbService.delete('keyPerformanceMetricImpact', id);
+  }
+
+  updateWithObservable(keyPerformanceMetricImpact: IdbKeyPerformanceMetricImpact): Observable<IdbKeyPerformanceMetricImpact> {
+    keyPerformanceMetricImpact.modifiedDate = new Date();
+    return this.dbService.update('keyPerformanceMetricImpact', keyPerformanceMetricImpact);
+  }
+
+  async asyncUpdate(keyPerformanceMetricImpact: IdbKeyPerformanceMetricImpact) {
+    keyPerformanceMetricImpact = await firstValueFrom(this.updateWithObservable(keyPerformanceMetricImpact));
+    await this.setKeyPerformanceMetricImpacts();
+  }
+
+  getByGuid(guid: string): IdbKeyPerformanceMetricImpact {
+    let keyPerformanceMetricImpacts: Array<IdbKeyPerformanceMetricImpact> = this.keyPerformanceMetricImpacts.getValue();
+    return keyPerformanceMetricImpacts.find(kpmImpact => {
+      return kpmImpact.guid == guid
+    });
+  }
+
+  getByNebGuid(nebGuid: string): Array<IdbKeyPerformanceMetricImpact> {
+    let keyPerformanceMetricImpacts: Array<IdbKeyPerformanceMetricImpact> = this.keyPerformanceMetricImpacts.getValue();
+    return keyPerformanceMetricImpacts.filter(kpmImpact => {
+      return kpmImpact.nebId == nebGuid
+    });
+  }
+
+  getByCompanyGuid(companyGuid: string): Array<IdbKeyPerformanceMetricImpact> {
+    let keyPerformanceMetricImpacts: Array<IdbKeyPerformanceMetricImpact> = this.keyPerformanceMetricImpacts.getValue();
+    return keyPerformanceMetricImpacts.filter(kpmImpact => {
+      return kpmImpact.companyId == companyGuid
+    });
+  }
+
+  getByKpiGuid(kpiGuid: string): Array<IdbKeyPerformanceMetricImpact> {
+    let keyPerformanceMetricImpacts: Array<IdbKeyPerformanceMetricImpact> = this.keyPerformanceMetricImpacts.getValue();
+    return keyPerformanceMetricImpacts.filter(kpmImpact => {
+      return kpmImpact.kpiGuid == kpiGuid
+    });
+  }
+
+  async updatePerformanceMetricBaseline(keyPerformanceIndicator: IdbKeyPerformanceIndicator, keyPerformanceMetric: KeyPerformanceMetric) {
+    let companyMetricImpacts: Array<IdbKeyPerformanceMetricImpact> = this.getByKpiGuid(keyPerformanceIndicator.guid);
+    for (let i = 0; i < companyMetricImpacts.length; i++) {
+      let metricImpact: IdbKeyPerformanceMetricImpact = companyMetricImpacts[i];
+      metricImpact.costAdjustment = (metricImpact.modificationValue * keyPerformanceMetric.costPerValue);
```

Comment:
Cannot verify the math here: costAdjustment (KPM impact) = modificationValue (KPM impact) * costPerValue (KPM). May need double check with @koay9f

Reply:
Currently the only math we have implemented is this method of cost adjustment:
`cost = units/yr * dollars/unit`
It's been pretty well established with everyone. When we have additional methods of calculating KPMs/NEBs this section will get updated accordingly